src/Application/Kernel.php

```diff
@@ -65,6 +66,7 @@ private function initContainer(): Container
             ->singleton(CommandBus::class, fn () => $container->get(GenericCommandBus::class))
             ->addInitializer(new ServerInitializer())
             ->addInitializer(new RequestInitializer())
+            ->addInitializer(new ResponseSenderInitializer())
```

Comment:
You don't need to manually add it here, you could add `#[InitializedBy]` on the `ResponseSender` interface. The reason some initializers are listed here, is because these initializers don't initialize one specific class. They implement an additional `CanInitialize` interface, which adds more dynamic behaviour. This is needed in some cases like for example when you're initializing the current request.

There might be some more intializers in this list that can be refactored to the `#[InitializedBy]` approach btw, I'll look into that after this one is merged.